examples/deriving-type-classes.flix

```diff
@@ -1,67 +1,37 @@
-// A Suit type deriving an Eq and ToString instance
-enum Suit with Eq, ToString {
-    case Clubs
-    case Hearts
-    case Spades
-    case Diamonds
+/// We derive the type classes Eq, Order, and ToString for the enum Month
+enum Month with Eq, Order, ToString {
+    case January
+    case February
+    case March
+    case April
+    case May
+    case June
+    case July
+    case August
+    case September
+    case October
+    case November
+    case December
 }
 
-// A Rank type deriving an Eq and Order instance
-enum Rank with Eq, Order {
-    case Number(Int)
-    case Jack
-    case Queen
-    case King
-    case Ace
-}
+type alias Year = Int
+type alias Day = Int
 
-// A Card type deriving an Eq instance
-opaque type Card with Eq = (Rank, Suit) 
+/// The Date type derives the type classes Eq and Order
+opaque type Date with Eq, Order = (Year, Month, Day)
 
-// An instance of ToString for Ranks
-instance ToString[Rank] {
-    pub def toString(x: Rank): String = match x {
-        case Number(n) => "${n}"
-        case Jack      => "Jack"
-        case Queen     => "Queen"
-        case King      => "King"
-        case Ace       => "Ace"
-    }
+/// We implement our own instance of ToString for Date
+/// since we don't want the default "Date(1948, December, 10)"
+instance ToString[Date] {
+    pub def toString(x: Date): String =
+        let Date(y, m, d) = x;
+        "${d} ${m}, ${y}"
 }
 
-// An instance of ToString for Cards
-instance ToString[Card] {
-    pub def toString(x: Card): String = match x {
-        case Card(r, s) => "${r} of ${s}"
-    }
-}
+/// Thanks to the Eq and Order type classes, we can easily compare dates.
+def earlierDate(d1: Date, d2: Date): Date = Order.min(d1, d2)
```

Comment:
I believe it works, but do you really want people think that they should use this date implementation?

Reply:
Yeah maybe not...